File: app/src/main/java/com/kickstarter/mock/factories/MessageThreadsEnvelopeFactory.java
package com.kickstarter.mock.factories;

import com.kickstarter.services.apiresponses.MessageThreadsEnvelope;

import java.util.Collections;

import androidx.annotation.NonNull;

public final class MessageThreadsEnvelopeFactory {
  private MessageThreadsEnvelopeFactory() {}

  public static @NonNull MessageThreadsEnvelope messageThreadsEnvelope() {
    return MessageThreadsEnvelope.builder()
      .urls(MessageThreadsEnvelope.UrlsEnvelope.builder()
        .api(MessageThreadsEnvelope.UrlsEnvelope.ApiEnvelope.builder()
          .moreMessageThreads("http://kck.str/message_threads/more")
          .build())
        .build())
      .messageThreads(Collections.singletonList(MessageThreadFactory.messageThread()))
      .build();
  }
}

File: app/src/main/java/com/kickstarter/mock/factories/MessageThreadsEnvelopeFactory.kt
package com.kickstarter.mock.factories

import com.kickstarter.mock.factories.MessageThreadFactory.messageThread
import com.kickstarter.services.apiresponses.MessageThreadsEnvelope

object MessageThreadsEnvelopeFactory {
    @JvmStatic
    fun messageThreadsEnvelope(): MessageThreadsEnvelope {
        return MessageThreadsEnvelope.builder()
            .urls(
                MessageThreadsEnvelope.UrlsEnvelope.builder()
                    .api(
                        MessageThreadsEnvelope.UrlsEnvelope.ApiEnvelope.builder()
                            .moreMessageThreads("http://kck.str/message_threads/more")
                            .build()
                    )
                    .build()
            )
            .messageThreads(listOf(messageThread()))
            .build()
    }
}
